server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -841,16 +996,16 @@ public BigInteger parse(XContentParser parser, boolean coerce) throws IOExceptio
             }
 
             @Override
-            public Query termQuery(String field, Object value) {
+            public Query termQuery(String field, Object value, boolean hasDocValues, boolean isSearchable) {
                 if (hasDecimalPart(value)) {
                     return Queries.newMatchNoDocsQuery("Value [" + value + "] has a decimal part");
                 }
                 BigInteger v = parse(value, true);
-                return BigIntegerPoint.newExactQuery(field, v);
+                return rangeQuery(field, v, v, true, true, hasDocValues, isSearchable, null);
```

Comment:
Question please: why we are using `rangeQuery` here for a single term match?

Reply:
@reta `BigIntegerPoint.newExactQuery` internally uses the same `rangeQuery` as `BigInteger.rangeQuery`, see -- https://github.com/harshavamsi/lucene/blob/8ac26737913d0c1555019e93bc6bf7db1ab9047e/lucene/sandbox/src/java/org/apache/lucene/sandbox/document/BigIntegerPoint.java#L174

Reply:
@reta to be consistent I have updated this. But the functionality exactly follows `UnsignedLong.rangeQuery`, do you still think that we need to do this?

Reply:
> @reta to be consistent I have updated this. But the functionality exactly follows `UnsignedLong.rangeQuery`, do you still think that we need to do this?

thanks @harshavamsi , why we don't use `SortedUnsignedLongDocValuesSetQuery`?

Reply:
Ah I now see what you mean. I added a `newSlowExactQuery` to the set query.